src/AElf.Contracts.Election/ElectionContract_Election.cs

```diff
@@ -209,14 +209,14 @@ public override Empty Vote(VoteMinerInput input)
             Assert(State.CandidateInformationMap[input.CandidatePublicKey].IsCurrentCandidate,
                 "Candidate quited election.");
 
-            var lockDays = (input.EndTimestamp - Context.CurrentBlockTime.ToTimestamp()).ToTimeSpan().TotalDays;
+            var lockDays = (input.EndTimestamp - Context.CurrentBlockTime.ToTimestamp()).Days();
             Assert(lockDays >= State.MinimumLockTime.Value,
                 $"Invalid lock time. At least {State.MinimumLockTime.Value} {(TimeUnit) State.BaseTimeUnit.Value}");
             Assert(lockDays <= State.MaximumLockTime.Value,
                 $"Invalid lock time. At most {State.MaximumLockTime.Value} {(TimeUnit) State.BaseTimeUnit.Value}");
 
             State.LockTimeMap[Context.TransactionId] =
-                GetTimeSpan(input.EndTimestamp.ToDateTime(), Context.CurrentBlockTime);
+                GetTimeSpan(input.EndTimestamp, Context.CurrentBlockTime.ToTimestamp());
```

Comment:
directly change the context interface

Reply:
To confirm, I will change Context.CurrentBlockTime type from DateTime to Timestamp?